components/suggest/src/db.rs

```diff
@@ -258,7 +279,9 @@ impl<'a> SuggestDao<'a> {
                     }
                 }
             }
-        )?.into_iter().flatten().collect())
+        )?.flat_map(Result::transpose).collect::<Result<_>>()?;
```

Comment:
This transformation filters out the `None`s from our `query_and_then(...)` iterator, then accumulates the remaining suggestions into a vector. It leans on a few iterator helpers to do this:

* `transpose` turns a `Result<Option<Suggestion>>` into an `Option<Result<Suggestion>>`.
* An `Option` is an iterator—it yields zero elements if it's `None`, and one if it's `Some`—so we can use it with `flat_map` to unwrap optionals. This turns our iterator of `Option<Result<Suggestion>>` into an iterator of `Result<Suggestion>`.
* `collect::<Result<_>>` accumulates the iterator of `Result<Suggestion>` into a `Result<Vec<Suggestion>>`.